lib/foreman_azure_rm/engine.rb

```diff
@@ -34,11 +34,18 @@ class Engine < ::Rails::Engine
     end
 
     config.to_prepare do
+
+      # pry here and implement .class_eval here
       require 'azure_mgmt_resources'
       require 'azure_mgmt_network'
       require 'azure_mgmt_storage'
       require 'azure_mgmt_compute'
 
+      ::Image.class_eval do
+        validates :uuid, format: { with: /\A((marketplace|custom|gallery):\/\/)([A-Za-z_\-]+)[\^:A-Za-z_\-]+[\^:A-Za-z_\-]+[\^:A-Za-z_\-]+/,
```

Comment:
Theoretically can be a bit shorter regex:

``` regex
^((marketplace|custom|gallery):\/\/)[^:]+(:[^:]+:[^:]+:[^:]+)?$
```

Meaning: Has to begin with `marketplace`, `custom` or `gallery`. Then `://`, then something that is not `:`, and then optionally `:<letters>:<letters>:<letters>`

Reply:
I tried implementing this regex, but it doesn't work for me.